### PR TITLE
documentation updates

### PR DESCRIPTION
**Pull request by porres**
documentation updates as usual

Closes: https://github.com/pure-data/pure-data/issues/1525

Closes: https://github.com/pure-data/pure-data/pull/1568

closes https://github.com/pure-data/pddp/issues/6

closes https://github.com/pure-data/pure-data/issues/1578

closes https://github.com/pure-data/pure-data/pull/1517 

closes https://github.com/pure-data/pddp/issues/56

closes https://github.com/pure-data/pure-data/issues/1527

### CODE REVIEW

File: doc/5.reference/fudi-format-parse-help.pd [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#X obj 356 398 list;
+#X listbox 356 422 24 0 0 0 - - - 0;
+#X listbox 356 281 76 0 0 0 - - - 0;
+#X obj 356 338 fudiparse -u;
+#X obj 375 307 print packet;
+#X msg 356 195 1 2 3 foo 5;
+#X text 20 75 FUDI stands for "Fast Universal Digital Interface" and
+is a networking protocol used by Pd. The fudiformat object makes FUDI
+messages suitable for sending over the network via netsend (in UDP
+mode). Incoming messages are output as FUDI messages \, byte by byte.
```
Comment (umlaeute) [inline doc/5.reference/fudi-format-parse-help.pd:38]: shouldn't that read "in *binary* mode"?

Comment (porres) [reply doc/5.reference/fudi-format-parse-help.pd:38]: instead of "byte by byte", right?

Comment (umlaeute) [reply doc/5.reference/fudi-format-parse-help.pd:38]: no, instead of "in UDP mode"

**Comment by porres** [comment]
Just added references to ALL help patches, all but "file"! I had this in a separate branch and just merged it into this documentation branch of mine.

**Comment by umlaeute** [comment]
great work! thanks for spending so much time.

some issues i've seen so far:
- https://github.com/pure-data/pure-data/issues/1527
- i think that unlike the original pddp `META` block, the reference information is not machine parseable, which i think is a pity, given the amount of work that went into the reference sections.
- while you are there, would it be possible to change the commit message for the last commit :wink:  i know it's fantastic when you finished such a big block of work - but the commit message is not really the place for todo lists. if you feel strong about it, you can safely ignore this last remark :stuck_out_tongue:

**Comment by porres** [comment]
> unlike the original pddp META block, the reference information is not machine parseable, which i think is a pity, given the amount of work that went into the reference sections.

I don't really understand what this means and what's goal. I know there's a [pd meta] standard used and seen in purr data and some externals, but I never got what this was supposed to be useful for...

I can put more work into it anyway... and include such a subpatch inside the the references?

**Comment by porres** [comment]
I can ignore this compilation error/warning, right?

**Comment by umlaeute** [comment]
> I don't really understand what this means and what's goal. I know there's a [pd meta] standard used and seen in purr data and some externals, but I never got what this was supposed to be useful for...

it's useful for automated information extraction.
e.g. it could be used 
- for automatic tooltips for iolets (i think purr data does that),
- or for creating an "online" (as in: HTML) documentation from the help -patches (that is always up-to-date)

**Comment by umlaeute** [comment]
~~~command
$ codespell doc/5.reference/ extra/*/*.pd
doc/5.reference/vu-help.pd:596: andd ==> and
doc/5.reference/vu-help.pd:612: defautl ==> default
doc/5.reference/vslider-help.pd:506: erros ==> errors
doc/5.reference/vradio-help.pd:511: erros ==> errors
doc/5.reference/trigger-help.pd:107: usefull ==> useful
doc/5.reference/toggle-help.pd:492: erros ==> errors
doc/5.reference/toggle-help.pd:624: andd ==> and
doc/5.reference/timer-help.pd:25: preceeded ==> preceded, proceeded
doc/5.reference/threshold~-help.pd:80: incomming ==> incoming
doc/5.reference/text-object-help.pd:649: preceeded ==> preceded, proceeded
doc/5.reference/text-object-help.pd:723: preceeded ==> preceded, proceeded
doc/5.reference/text-object-help.pd:747: lenght ==> length
doc/5.reference/text-object-help.pd:901: curren ==> current
doc/5.reference/tabosc4~-help.pd:100: frequenct ==> frequency, frequent
doc/5.reference/spigot-help.pd:38: openned ==> opened
doc/5.reference/set-help.pd:75: andd ==> and
doc/5.reference/route-help.pd:125: mesages ==> messages
doc/5.reference/route-help.pd:188: exeptions ==> exceptions
doc/5.reference/qlist-help.pd:89: supresses ==> suppresses
doc/5.reference/pow~-help.pd:30: agains ==> against, again
doc/5.reference/plot-help.pd:132: enabel ==> enable
doc/5.reference/phasor~-help.pd:64: frequenct ==> frequency, frequent
doc/5.reference/pdcontrol-help.pd:11: visibilty ==> visibility
doc/5.reference/pd-messages.pd:13: beccause ==> because
doc/5.reference/pd-messages.pd:17: compatibilty ==> compatibility
doc/5.reference/pd-messages.pd:18: compatibilty ==> compatibility
doc/5.reference/pd-messages.pd:39: miliseconds ==> milliseconds
doc/5.reference/pd-messages.pd:70: milisecond ==> millisecond
doc/5.reference/pd-messages.pd:267: Sructures ==> Structures
doc/5.reference/pd-messages.pd:383: preceed ==> precede, proceed
doc/5.reference/pack-help.pd:86: acccording ==> according
doc/5.reference/osc~-help.pd:57: frequenct ==> frequency, frequent
doc/5.reference/osc-format-parse-help.pd:88: adresses ==> addresses
doc/5.reference/osc-format-parse-help.pd:93: interger ==> integer
doc/5.reference/osc-format-parse-help.pd:94: sring ==> string
doc/5.reference/netsend-help.pd:149: recieved ==> received
doc/5.reference/nbx-help.pd:437: erros ==> errors
doc/5.reference/midi-help.pd:186: supresses ==> suppresses
doc/5.reference/midi-help.pd:236: supresses ==> suppresses
doc/5.reference/midi-help.pd:238: supresses ==> suppresses
doc/5.reference/midi-help.pd:276: supresses ==> suppresses
doc/5.reference/midi-help.pd:315: supresses ==> suppresses
doc/5.reference/midi-help.pd:321: programm ==> program, programme
doc/5.reference/midi-help.pd:357: supresses ==> suppresses
doc/5.reference/midi-help.pd:405: supresses ==> suppresses
doc/5.reference/metro-help.pd:93: preceeded ==> preceded, proceeded
doc/5.reference/message-help.pd:136: sintax ==> syntax
doc/5.reference/message-help.pd:310: wihtout ==> without
doc/5.reference/math-help.pd:78: agains ==> against, again
doc/5.reference/list-help.pd:255: incomming ==> incoming
doc/5.reference/list-help.pd:746: fisrt ==> first
doc/5.reference/list-help.pd:749: fisrt ==> first
doc/5.reference/key-input-help.pd:16: ouput ==> output
doc/5.reference/int-help.pd:65: intput ==> input
doc/5.reference/int-help.pd:114: abreviation ==> abbreviation
doc/5.reference/inlet-outlet-help.pd:208: singal ==> signal, single
doc/5.reference/hslider-help.pd:518: erros ==> errors
doc/5.reference/hradio-help.pd:511: erros ==> errors
doc/5.reference/gui-boxes-help.pd:248: preceeded ==> preceded, proceeded
doc/5.reference/gui-boxes-help.pd:351: Ths ==> The, this
doc/5.reference/expr-help.pd:667: Accoustic ==> Acoustic
doc/5.reference/expr-help.pd:731: lke ==> like
doc/5.reference/drawtext-help.pd:53: wolrd ==> world
doc/5.reference/draw-shapes-help.pd:33: visibilty ==> visibility
doc/5.reference/draw-shapes-help.pd:35: whith ==> with
doc/5.reference/draw-shapes-help.pd:36: heigh ==> height, high
doc/5.reference/delay-help.pd:29: preceeded ==> preceded, proceeded
doc/5.reference/declare-help.pd:88: neeed ==> need
doc/5.reference/clone-help.pd:17: circularly ==> circular
doc/5.reference/clone-help.pd:71: circularly ==> circular
doc/5.reference/clip-help.pd:40: incomming ==> incoming
doc/5.reference/bng-help.pd:534: erros ==> errors
doc/5.reference/bng-help.pd:610: andd ==> and
doc/5.reference/bang-help.pd:19: abreviation ==> abbreviation
doc/5.reference/array-object-help.pd:76: arrray ==> array
doc/5.reference/array-object-help.pd:336: arrray ==> array
doc/5.reference/array-object-help.pd:414: arrray ==> array
doc/5.reference/array-object-help.pd:482: arrray ==> array
doc/5.reference/array-object-help.pd:579: betwee ==> between
doc/5.reference/array-object-help.pd:748: minmum ==> minimum
doc/5.reference/array-object-help.pd:794: acccess ==> access
doc/5.reference/array-object-help.pd:799: acccess ==> access
doc/5.reference/acoustics~-help.pd:83: incomming ==> incoming
doc/5.reference/acoustics-help.pd:60: incomming ==> incoming
extra/lrshift~/lrshift~-help.pd:30: versin ==> version
~~~

**Comment by porres** [comment]
I think I'm done with fixing typos, you can do one more round check. 

I didn't change "circularly" as I think this is correct.

**Comment by porres** [comment]
Just installed codespell here on my mac as well, what a divine tool :) still have left some corrections out:

Alexandres-MacBook-Air:pure-data porres$ codespell doc/5.reference/ extra/*/*.pd
doc/5.reference/declare-help.pd:88: neeed ==> need
doc/5.reference/trigger-help.pd:107: usefull ==> useful
doc/5.reference/draw-shapes-help.pd:58: whith ==> with
doc/5.reference/pd-messages.pd:70: milisecond ==> millisecond
doc/5.reference/array-object-help.pd:579: betwee ==> between
doc/5.reference/array-object-help.pd:748: minmum ==> minimum
doc/5.reference/array-object-help.pd:794: acccess ==> access
doc/5.reference/array-object-help.pd:799: acccess ==> access
doc/5.reference/route-help.pd:188: exeptions ==> exceptions
doc/5.reference/clone-help.pd:17: circularly ==> circular
doc/5.reference/clone-help.pd:71: circularly ==> circular

**Comment by porres** [comment]
>I didn't change "circularly" as I think this is correct.

fixed the typos above but still kept "circularly"

**Comment by porres** [comment]
Should I worry about this failed check? Doesn't seem like something I can fix...

**Comment by porres** [comment]
Codespell is giving me more typos when going over the whole 'doc' folder, here it is:

Last login: Tue Mar 22 22:40:51 on ttys001
Alexandres-MacBook-Air:doc porres$ codespell
./2.control.examples/16.more.arrays.pd:30: automaticallly ==> automatically
./1.manual/x4.htm:494: preferr ==> prefer
./1.manual/x4.htm:553: distiction ==> distinction
./1.manual/x4.htm:559: overriden ==> overridden
./1.manual/x4.htm:560: instace ==> instance
./1.manual/x4.htm:570: overriden ==> overridden
./1.manual/x4.htm:596: disribution ==> distribution
./1.manual/x5.htm:73: visiblity ==> visibility
./1.manual/x5.htm:122: Sur ==> Sure, sir
./1.manual/x5.htm:376: crasher ==> crash
./1.manual/x5.htm:387: crasher ==> crash
./1.manual/x2.htm:1126: wit ==> with
./1.manual/x3.htm:509: MSDOS ==> MS-DOS
./3.audio.examples/F12.paf.pd:110: wit ==> with
./7.stuff/soundfile-tools/3.phase.vocoder.pd:471: lenght ==> length
./5.reference/clone-help.pd:17: circularly ==> circular
./5.reference/clone-help.pd:71: circularly ==> circular
./8.topics/expr.html:164: inlcudes ==> includes
Alexandres-MacBook-Air:doc porres$ 


extra folder also gives me:

Alexandres-MacBook-Air:extra porres$ codespell
./bob~/bob~.c:212: varable ==> variable
./pd~/pd~.c:1200: nin ==> inn, min, bin, nine
./pd~/pd~.c:1207: nin ==> inn, min, bin, nine
Alexandres-MacBook-Air:extra porres$

**Comment by porres** [comment]
fixed typos above, but not all as they're not all actual typos... I kept (again) circularly, as well as others like 'to wit' in "3.audio.examples/F12.paf.pd" (I don't understand this but it seems it is correct and on purpose).

**Comment by porres** [comment]
even though documentation is taken care of, doing a "codespell" over the whole code in repository gives us lots of stuff that might not hurt to fix them...

**Comment by umlaeute** [comment]
no need to worry about the mingw build failure; i re-enabled it to see if the cache issue was gone, but apparently not. i've disabled it for now. in general you should be able to just check the "Details" of a failed job to see whether the failure relates to your changes or something else.

> `circularly`

yes, this seems to be a false positive

> `to wit`

i don't understand it either :-) let's assume it's a false positive

> doing a "codespell" over the whole code in repository gives us lots of stuff that might not hurt to fix them...

actually most of it *will* hurt when fixing it...

- codespell does an English spell-checking; practically all findings in `po/` are false positives from other languages
- we do a verbatim import of `portaudio` upstream sources; we should *not* fix spelling errors in our copy (but feel free to do a PR with [upstream](https://github.com/PortAudio/portaudio), which will magically fix the mistakes whenever we do the next import); same for `portmidi`
- the remaining spelling mistakes in `src/` and `tcl/` (that haven't been fixed yet with 31a4c70ae17144ddb948a8c31b62ae76b9ddf8f9) are all variable names rather than English words, which are spelt like they are for a reason (afaict only the `.find.searchin` object name is debatable)

**Comment by umlaeute** [comment]
i know, I'm always nitpicking... but:

- i think the `<= click` label for each and every reference-patch is just noise that doesn't add any value. there are other sub-patches in the helps that do not have an extra "<= click" comment, and I assume that people who have a basic understanding of how Pd works will soon understand that they can open the `[pd reference]` to get the reference.
- sometimes the iolet reference uses two distinct comments (e.g. `float -` and `sets variable value.`), and sometimes only a single one (e.g. `symbol - sets the value name.`). In some reference-patches you can see both (e.g. in value-help.pd). I know it *looks* the same but it would be somewhat nice if they actually were the same (e.g. for copy-editors like me, who wonder why they only copied half of the text in some cases)
- there's a typo in text-help.pd where it should read "and have no inlet/outlets" rather than ", has no inlet/outlets" (it should be plural)
- 3.audio.examples/F12.paf.pd has `~*` instead of `*~` (in a comment, so there's no error)

**Comment by porres** [comment]
> sometimes the iolet reference uses two distinct comments (e.g. float - and sets variable value.), and sometimes only a single one (e.g. symbol - sets the value name.). In some reference-patches you can see both (e.g. in value-help.pd). I know it looks the same but it would be somewhat nice if they actually were the same (e.g. for copy-editors like me, who wonder why they only copied half of the text in some cases)

I didn't really get this one. There was an error for the outlet description, I fixed it.

**Pull request merged**